src/Tizen.Multimedia.Vision/MediaVision/DeepLearningFaceRecognier.cs

```diff
@@ -0,0 +1,176 @@
+        /// <since_tizen> 10 </since_tizen>
+        public DeepLearningFaceRecognizer()
+        {
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.FaceRecognition);
+
+            InteropFace.Create(out _handle).Validate("Failed to create face recognizer");
+
+            try
+            {
+                InteropFace.Prepare(_handle).Validate("Failed to prepare face recognizer");
+            }
+            catch (Exception e)
+            {
+                Log.Error(MediaVisionLog.Tag, e.ToString());
+                InteropFace.Destroy(_handle);
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DeepLearningFaceRecognizer class.
+        /// </summary>
+        ~DeepLearningFaceRecognizer()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Registers face data to internal database.
+        /// </summary>
+        /// <param name="source">The face data to register.</param>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The DeepLearningFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void RegisterFace(MediaVisionSource source, string label)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Register(_handle, source.Handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Unregisters face data from internal database.
+        /// </summary>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The DeepLearningFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void UnregisterFace(string label)
+        {
+            ValidateNotDisposed();
+
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Unregister(_handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Recognizes input face from internal face datas and returns its label.
```

Comment:
"data" is special case word (singular mass noun/uncountable), you cannot write "datas", consider rewriting it to something like:
"Recognizes a face in <paramref name="source"/> by finding the closest match among the registered faces and returns the label of the found face."
Not sure how it works. Would it look for a face in the source, or should the source be already cropped and contain face only?

Reply:
Changed as your comment.